eodag/rest/utils.py

```diff
@@ -82,6 +84,22 @@
 
 eodag_api = eodag.EODataAccessGateway()
 
+max_len = int(os.getenv("EODAG_SEARCH_RESULTS_CACHE_MAXLEN", DEFAULT_DEQUE_MAX_LEN))
+cached_search_results: Deque = collections.deque(maxlen=max_len)
```

Comment:
Deque type should be defined. Example: Deque[Dict[str, Any]].
A dataclass would be more suited here. Something like:

@dataclass
class CacheRecord:
    """cache record"""
    provider: str
    product_type: str
    results: SearchResult
    
    with: cached_search_results: Deque[SearchRecord] = collections.deque(maxlen=max_len)

Reply:
done